Лаб. 2 6412 Шестериков Борис

BarMan555 opened the pull request:
**ФИО:** Шестериков Борис
**Номер группы:** 6412 
**Номер лабораторной:** 2  
**Номер варианта:** 54
**Краткое описание предметной области:** Поликлиника
**Краткое описание добавленных фич:** Добавлен проект с in-memory репозиториями для всех доменных сущностей, которые используют seeders для заполнения тестовыми данными и поддерживают базовые CRUD-операци
Реализован сервисный слой с классами, наследующими общий интерфейс IApplicationService и ILibraryAnalyticsService. 
Для маппинга между доменными моделями и DTO используется библиотека AutoMapping
Создан базовый контроллер CrudControllerBase, предоставляющий стандартные методы API с обработкой ошибок и логированием.

danlla reviewed (changes requested):
Решение билдится с ошибкой (что также видно в не пройденных тестах), это нужно поправить, полное ревью сделаю после исправления
А так если сразу то, что бросается в глаза: 

1. 8 варнингов
2. Не везде file scoped namespace
3. Нет запросов из задания в контроллерах
4. Нарушения в наименованиях локальных переменных, некоторые из них в `snake_case`
5. Dto модели соответствуют моделям в в `Domain`, зачем они тогда нужны?
6. Все что находится в классе `Mapper` можно сделать используя функционал автомаппера

danlla commented on AdmissionCommittee/Interfaces/IRepository.cs:
```diff
@@ -0,0 +1,39 @@
+﻿namespace AdmissionCommittee.Domain.Interfaces;
+
+public interface IRepository<TEntity, Tkey>
```
`TKey`

danlla commented on AdmissionCommittee/Repositories/ApplicantRepository.cs:
```diff
@@ -0,0 +1,77 @@
+    public ApplicantRepository(List<Applicant> applicants)
+    {
+        _applicants = applicants;
+        //    _applicants.Add(new() { Id = 0, BirthdayDate = new DateTime(2005, 1, 18), City = "Samara", Country = "Russia", FullName = "Vladimir Vladimirovich" });
+        //    _applicants.Add(new() { Id = 1, BirthdayDate = new DateTime(2002, 2, 9), City = "Samara", Country = "Russia", FullName = "Andrew Viktorovich" });
+        //    _applicants.Add(new() { Id = 2, BirthdayDate = new DateTime(2005, 7, 8), City = "Vladivostok", Country = "Russia", FullName = "Vitaliy Vitalivich" });
+        //    _applicants.Add(new() { Id = 3, BirthdayDate = new DateTime(2004, 1, 13), City = "Samara", Country = "Russia", FullName = "Michail Michailovich" });
+        //    _applicants.Add(new() { Id = 4, BirthdayDate = new DateTime(2004, 6, 2), City = "Saints-Petersburg", Country = "Russia", FullName = "Veronika Igorevna" });
+        //    _applicants.Add(new() { Id = 5, BirthdayDate = new DateTime(2004, 2, 12), City = "Samara", Country = "Russia", FullName = "Ivan Ivanov" });
+        //    _applicants.Add(new() { Id = 6, BirthdayDate = new DateTime(2005, 2, 22), City = "Vladivostok", Country = "Russia", FullName = "Danila Danilovich" });
+        //    _applicants.Add(new() { Id = 7, BirthdayDate = new DateTime(2001, 2, 2), City = "Samara", Country = "Russia", FullName = "Maria Olegovna" });
+        //    _applicants.Add(new() { Id = 8, BirthdayDate = new DateTime(2002, 1, 4), City = "Moscow", Country = "Russia", FullName = "Sergey Sergeevich" });
+        //    _applicants.Add(new() { Id = 9, BirthdayDate = new DateTime(2004, 12, 1), City = "Vladivostok", Country = "Russia", FullName = "Vladimir Vladimirov" });
```
Убрать комментарии

danlla commented on AdmissionCommittee/Repositories/ApplicantRepository.cs:
```diff
@@ -0,0 +1,77 @@
+    /// <returns>If item not found return false, else true</returns>
+    public bool UpdateById(Applicant newItem, int id)
+    {
+        var item_id = _applicants.FindIndex(a => a.Id == id);
```
Нарушение наименования

Тут можно сразу получить объект с помощью метода `GetById` проверить его на null и работать дальше с ним

В остальных репозиториях так же

danlla commented on AdmissionCommittee.Server/ReaderCSV.cs:
```diff
@@ -0,0 +1,106 @@
+                    string[] rows = textFieldParser.ReadFields();
+                    Applicant applicant = new Applicant
+                    {
+                        Id = Int32.Parse(rows[0]),
```
`int`

везде где `Int32` также поменять

danlla commented on AdmissionCommittee.Application/AutoMapperP.cs:
```diff
@@ -0,0 +1,18 @@
+﻿using AdmissionCommittee.Application.DTO;
+using AdmissionCommittee.Domain.Models;
+using AutoMapper;
+
+public class AutoMapperP : Profile
+{
+    /// <summary>
```
Нет namespace

danlla commented on AdmissionCommittee/Repositories/ExamResultRepository.cs:
```diff
@@ -0,0 +1,66 @@
+
+public class ExamResultRepository : IRepository<ExamResult, int>
+{
+    private static List<ExamResult> _eResults = [];
```
`_examResults`

danlla reviewed (changes requested):
После всех правок проверить, что error list для всего проекта пустой

danlla commented on AdmissionCommittee.Server/ReaderCSV.cs:
```diff
@@ -0,0 +1,106 @@
+        {
+            var applicants = new List<Applicant>();
+            using var textFieldParser = new TextFieldParser(fileName);
+            {
+                textFieldParser.TextFieldType = FieldType.Delimited;
+                textFieldParser.SetDelimiters(";");
+                while (!textFieldParser.EndOfData)
+                {
+                    var rows = textFieldParser.ReadFields();
+                    Applicant applicant = new Applicant
+                    {
+                        Id = int.Parse(rows[0]),
+                        BirthdayDate = DateTime.ParseExact(rows[1], "dd.MM.yyyy", CultureInfo.InvariantCulture),
+                        City = rows[2],
+                        Country = rows[3],
+                        FullName = rows[4]
+                    };
+                    applicants.Add(applicant);
+                }
+            }
```
На 13 и 29 строчках не нужны скобки

В методах ниже так же

danlla commented on AdmissionCommittee/Repositories/ApplicantRepository.cs:
```diff
@@ -0,0 +1,72 @@
+    /// <param name="newItem"><see cref="Applicant"/> item</param>
+    public void Add(Applicant newItem)
+    {
+        var count = GetAll().Count - 1;
+        var newId = _applicants[count].Id + 1;
```
Работу с id лучше вынести в переменную в классе, потому что даже при таком подходу будут дубли

В остальных репозиториях так же

danlla reviewed (changes requested):
Error list до сих пор содержит предупреждения и сообщения об улучшении кода

danlla commented on AdmissionCommittee/Repositories/ApplicantRepository.cs:
```diff
@@ -0,0 +1,74 @@
+    /// <param name="newItem"><see cref="Applicant"/> item</param>
+    public void Add(Applicant newItem)
+    {
+        _applicantsCount++;
+        var newId = _applicantsCount;
+        newItem.Id = newId;
```
```csharp
newItem.Id = ++_applicantsCount;
```

в остальных репозиториях так же

danlla commented on AdmissionCommittee.Server/Controllers/ApplicantController.cs:
```diff
@@ -0,0 +1,104 @@
+    [HttpGet]
+    public ActionResult<IEnumerable<Applicant>> Get()
+    {
+        var applicationDto = repository.GetAll();
+
+        if (applicationDto == null) return NotFound();
```
Это не `applicantionDto` и эта переменная не может быть `null`

В остальных контроллерах так же

danlla commented on AdmissionCommittee.Server/ReaderCSV.cs:
```diff
@@ -0,0 +1,111 @@
+                var rows = textFieldParser.ReadFields();
+                if (rows != null)
+                {
+                    Applicant applicant = new Applicant
```
```csharp
var applicant = new Applicant
{
    Id = int.Parse(rows[0]),
    BirthdayDate = DateTime.ParseExact(rows[1], "dd.MM.yyyy", CultureInfo.InvariantCulture),
    City = rows[2],
    Country = rows[3],
    FullName = rows[4]
};
```

danlla commented on AdmissionCommittee.Server/ReaderCSV.cs:
```diff
@@ -0,0 +1,111 @@
+                var rows = textFieldParser.ReadFields();
+                if (rows != null)
+                {
+                    Direction direction = new Direction
```
```csharp
var direction = new Direction
{
    Id = int.Parse(rows[0]),
    ApplicantId = int.Parse(rows[1]),
    SpecialityId = int.Parse(rows[2]),
    Priority = int.Parse(rows[3])
};
```

danlla commented on AdmissionCommittee.Server/ReaderCSV.cs:
```diff
@@ -0,0 +1,111 @@
+                var rows = textFieldParser.ReadFields();
+                if (rows != null)
+                {
+                    ExamResult eResult = new ExamResult
+                    {
+                        Id = int.Parse(rows[0]),
+                        ApplicantId = int.Parse(rows[1]),
+                        ExamName = rows[2],
+                        Result = int.Parse(rows[3]),
+                    };
+                    examResults.Add(eResult);
```
```csharp
var examResult = new ExamResult
{
    Id = int.Parse(rows[0]),
    ApplicantId = int.Parse(rows[1]),
    ExamName = rows[2],
    Result = int.Parse(rows[3]),
};
examResults.Add(examResult);
```

danlla commented on AdmissionCommittee.Server/ReaderCSV.cs:
```diff
@@ -0,0 +1,111 @@
+                var rows = textFieldParser.ReadFields();
+                if (rows != null)
+                {
+                    Speciality speciality = new Speciality
+                    {
+                        Id = int.Parse(rows[0]),
+                        Number = rows[1],
+                        Name = rows[2],
+                        Faculity = rows[3],
+                    };
```
```csharp
var speciality = new Speciality
{
    Id = int.Parse(rows[0]),
    Number = rows[1],
    Name = rows[2],
    Faculity = rows[3],
};
```

danlla commented on AdmissionCommittee.Server/Controllers/ApplicantController.cs:
```diff
@@ -0,0 +1,103 @@
+    [HttpGet]
+    public ActionResult<IEnumerable<Applicant>> Get()
+    {
+        var applicant = repository.GetAll();
```
Это и не `applicant`)

Можно просто сделать вот так:
```csharp
return Ok(repository.GetAll())
```

в остальных контроллерах так же

BarMan555 committed:
Make sln and 2 prj(classes and tests)

Make solution and 2 project. First for classes of models, second for tests

BarMan555 committed:
Make full structur for PolyclinicSolution

Make full structur for PolyclinicSolution

BarMan555 committed:
Rename projects and folders + add required for properties

Polyclinic -> Hospital

BarMan555 committed:
Make xUnit tests and make fixtures data

BarMan555 committed:
Add summary

BarMan555 committed:
Add extra summary

BarMan555 committed:
Rename Folders

BarMan555 committed:
Cleap up code, remake specialization object, rename fun in tests

BarMan555 committed:
Add DateTime.Now in test

BarMan555 committed:
Add patient interface repository

Create fold Repositories for interfaces of repositories. Add patient interface repository

BarMan555 committed:
Add infrastructure project with InMemoryPatientRepository.cs repository

Add infrastructure project with InMemoryPatientRepository.cs repository

BarMan555 committed:
Add seeder for Patient Repository

Add seeder for Patient Repository

BarMan555 committed:
Add application project.

Add application project: Patient DTO and Patient Service class

BarMan555 committed:
Add interface for doctor and appointnent repositories

Add interface for doctor and appointnent repositories

BarMan555 committed:
Add realization for doctor and appointment repository.

Add realization for doctor and appointment repository.

BarMan555 committed:
Make seeders for doctor and appointment repositories.

Make seeders for doctor and appointment repositories.

BarMan555 committed:
Add DTO class for doctor and appointment classes.

Add DTO class for doctor and appointment classes.

BarMan555 committed:
Add service class for doctor and appointment repositories.

Add service class for doctor and appointment repositories.

BarMan555 committed:
Make repositories fixture for testing sevices.

Make repositories fixture for testing sevices.

BarMan555 committed:
Relocataion fixture folder to test project.

Relocataion fixture folder to test project.

BarMan555 committed:
Add function for working with data and for testing.

Add function for working with data and for testing.

BarMan555 committed:
Add file with repositories and services tests.

Add file with repositories and services tests.

BarMan555 committed:
Add interfaces of seeder and repository. Add abstract class with realization of repository interface.

Make ISeeder.cs, which include basic entity for all other seeders. Make IRepository.cs, which include basic entity for all others repositories. Make InMemoryRepository.cs abstract class which implements basich method from IRepository.cs. Make classes InMemoryPatientRepository.cs, InMemoryDoctorRepository.cs and InMemoryAppointmentRepository.cs inherteds to abstract class InMemoryRepository.cs. Delete IPatientRepository.cs, IDoctorRepository.cs and IAppointmentRepository.cs becaouse they don't need. Classes PatientSeeder.cs, DoctorSeeder.cs and AppointmentSeeder.cs implement ISeeder.cs

BarMan555 committed:
Add project for services contracts

BarMan555 committed:
Make project with API Web Core interface.

Make project with API Web Core interface.

BarMan555 committed:
Add to api project.

Add to api project.

BarMan555 committed:
Make interfacies for sevice classes.

Make interfacies for sevice classes. IApplicationService.cs fro CRUD functions. ILibraryAnalyticsService.cs for analytics functions.

BarMan555 committed:
Add auto mapping in projects.

Add auto mapping in projects.

BarMan555 committed:
Add small fixes.

Add small fixes.

BarMan555 committed:
Add service which implements analytics interface.

Add service which implements analytics interface.

BarMan555 committed:
Add summary.

Add summary.

BarMan555 committed:
Add summary.

Add summary.

BarMan555 committed:
Add summary.

Add summary.

BarMan555 committed:
Why not?

BarMan555 committed:
Make README.md for my project.

Make README.md for my project.

BarMan555 committed:
Fix README.md

BarMan555 committed:
Fix README.md

BarMan555 committed:
Fix README.md

alxmcs commented on Hospital/.idea/.idea.Hospital/.idea/.gitignore:
Служебные файлы ide добавить в гитигнор, чтобы они не попадали в репозиторий

alxmcs commented on Hospital/.idea/.idea.Hospital/.idea/indexLayout.xml:
Служебные файлы ide добавить в гитигнор, чтобы они не попадали в репозиторий

alxmcs commented on Hospital/.idea/.idea.Hospital/.idea/vcs.xml:
Служебные файлы ide добавить в гитигнор, чтобы они не попадали в репозиторий

alxmcs commented on Hospital/Hospital.Api/Controllers/AnalyticController.cs:
```diff
@@ -0,0 +1,142 @@
+    /// <summary>
+    /// Helper method for consistent logging and error handling.
+    /// </summary>
+    private ActionResult Logging(string method, Func<ActionResult> action)
```
<img width="720" height="143" alt="Image" src="https://github.com/user-attachments/assets/e1256cf6-8282-42c2-ae83-f1630cff0141" />

alxmcs commented on Hospital/Hospital.Api/Controllers/AnalyticController.cs:
```diff
@@ -0,0 +1,142 @@
+    [HttpGet("patients-older-than")]
+    [ProducesResponseType(typeof(List<PatientDto>), StatusCodes.Status200OK)]
+    [ProducesResponseType(500)]
+    public ActionResult<List<PatientDto>> GetPatientsOlderThaneWithMultipleDoctors([FromQuery] int age)
```
<img width="728" height="60" alt="Image" src="https://github.com/user-attachments/assets/24c693cd-2351-4f1f-9115-99db7bb85bcb" />

alxmcs replied:
<img width="668" height="99" alt="Image" src="https://github.com/user-attachments/assets/fce4e6a6-2794-4969-a73d-c7200dfedd2e" />

alxmcs commented on Hospital/Hospital.Api/Controllers/CrudBaseController.cs:
```diff
@@ -0,0 +1,99 @@
+    where TDto : class
+    where TKey : struct
+{
+    /// <summary>
+    /// Helper method for consistent logging and error handling.
+    /// </summary>
+    protected ActionResult Logging(string method, Func<ActionResult> action)
+    {
+        logger.LogInformation("START: {Method}", method);
+        try
+        {
+            var result = action();
+            var count = 0;
+            if (result is OkObjectResult okResult && okResult.Value != null)
+            {
+                if (okResult.Value is System.Collections.IEnumerable collection)
+                {
+                    count = collection.Cast<object>().Count();
+                }
+                else count = 1;
+            }
+            logger.LogInformation("SUCCESS: {Method}. Found {Count} records.", method, count);
+            return result;
+        }
+        catch (Exception ex)
+        {
+            logger.LogError(ex, "ERROR: {Method} failed.", method);
+            return StatusCode(500, $"Server error: {ex.Message}");
+        }
+    }
```
О, этот уродский метод еще и не один раз повторяется

alxmcs commented on Hospital/Hospital.Api/Controllers/DoctorController.cs:
```diff
@@ -0,0 +1,15 @@
+/// <param name="logger">Logger for logging</param>
+[ApiController]
+[Route("api/[controller]")]
+public class DoctorController(IApplicationService<DoctorDto, int> service, ILogger<DoctorController> logger) 
```
А где маршруты для получения коллекций связанных сущностей?

alxmcs commented on Hospital/Hospital.Api/Controllers/PatientController.cs:
```diff
@@ -0,0 +1,15 @@
+/// <param name="logger">Logger for logging</param>
+[ApiController]
+[Route("api/[controller]")]
+public class PatientController(IApplicationService<PatientDto, int> service, ILogger<PatientController> logger) 
```
А где  маршруты для получения коллекций связанных сущностей?

alxmcs commented on Hospital/Hospital.Api/Controllers/AnalyticController.cs:
```diff
@@ -0,0 +1,142 @@
+[ApiController]
+[Route("api/[controller]")]
+public class AnalyticController(
+    ILibraryAnalyticsService analService, 
```
Поскольку я искренне надеюсь, что до метода `Logging` тебе помог додуматься стохастический попугай, ниже будут комментарии другого стохастического попугая к этому коду.
> Если ты додумался до него самостоятельно, то соболезную
<img width="750" height="222" alt="Image" src="https://github.com/user-attachments/assets/9aca7052-3d7f-48c2-aa8f-61195360495a" />

alxmcs commented on Hospital/Hospital.Api/Program.cs:
```diff
@@ -0,0 +1,40 @@
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var app = builder.Build();
```
Следующий pr, который мне пришлют без предварительного запуска кода, получит ноль.
<img width="800" height="175" alt="Image" src="https://github.com/user-attachments/assets/d6d0c417-5ad6-405c-bd1d-cf5bab3bb92f" />

alxmcs commented on Hospital/Hospital.Api/Controllers/CrudBaseController.cs:
```diff
@@ -0,0 +1,99 @@
+    [HttpGet]
+    [ProducesResponseType(200)]
+    [ProducesResponseType(500)]
+    public ActionResult<List<TDto>> GetAll() => Logging(nameof(GetAll), () => Ok(appService.GetAll()));
```
Расскажи мне пожалуйста, откуда потребитель апи будет брать все эти замечательные `TKey id`, если ты их в виде контрактов никогда не возвращаешь?

alxmcs commented on Hospital/Hospital.Application.Contracts/Dtos/AppointmentGetDto.cs:
Тут идентификатора сущности в контракте нет

alxmcs commented on Hospital/Hospital.Application.Contracts/Dtos/DoctorGetDto.cs:
И тут тоже нет

alxmcs commented on Hospital/Hospital.Application.Contracts/Dtos/PatientGetDto.cs:
И тут нет

alxmcs commented on Hospital/Hospital.Domain/Seeders/ISeeder.cs:
Я еще с момента первой лабы говорил, что сидер должен лежать (за неимением лучшего) в **доменном** проекте. Потому что в текущей конфигурации проектов, когда ты переведешь в 3 лабе хранение данных в бд, тебе придется тащить в зависимостях и этот ставший бесполезным проект, потому что сидеры объявлены и имплементированы только тут

alxmcs commented on Hospital/Hospital.Api/Hospital.Api.csproj:
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk.Web">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <Nullable>enable</Nullable>
+        <ImplicitUsings>enable</ImplicitUsings>
```
Документирование не включено

alxmcs commented on Hospital/Hospital.Application.Contracts/Hospital.Application.Contracts.csproj:
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
```
Документирование не включено

alxmcs commented on Hospital/Hospital.Application/Hospital.Application.csproj:
```diff
@@ -0,0 +1,18 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
```
Документирование не включено

alxmcs commented on Hospital/Hospital.Domain/Hospital.Domain.csproj:
```diff
@@ -0,0 +1,9 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
```
Документирование не включено

alxmcs commented on Hospital/Hospital.Infrastructure.InMemory/Hospital.Infrastructure.InMemory.csproj:
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
```
Документирование не включено

BarMan555 committed:
Update .gitignore

BarMan555 committed:
Add own interface for each service. Add new api methods.

BarMan555 committed:
Fix Dtos

BarMan555 committed:
Fix Seeders

BarMan555 committed:
Enable documentation

BarMan555 committed:
Fix controller and add mapper in Program.cs

alxmcs commented on Hospital/Hospital.Api/Controllers/PatientController.cs:
<img width="1422" height="999" alt="image" src="https://github.com/user-attachments/assets/570efaea-ffe8-43a3-a4b0-4a5baa3ce5fe" />
Я начинаю уставать повторять "хотя бы запусти код перед тем как попытаться его сдать"

alxmcs commented on Hospital/Hospital.Api/Controllers/DoctorController.cs:
<img width="1421" height="922" alt="image" src="https://github.com/user-attachments/assets/0dd09fbb-c83f-4e34-aabe-d661e5b0c783" />

alxmcs commented on Hospital/Hospital.Api/Controllers/AppointmentController.cs:
<img width="1428" height="860" alt="image" src="https://github.com/user-attachments/assets/27012424-f0dc-4952-a4b2-67d51bd115f1" />

alxmcs commented on Hospital/Hospital.Api/Controllers/AnalyticController.cs:
<img width="1427" height="984" alt="image" src="https://github.com/user-attachments/assets/b21ee10d-9726-470e-ba5d-e465d154c34e" />

alxmcs commented on Hospital/Hospital.Api/Program.cs:
```diff
@@ -0,0 +1,53 @@
+
+builder.Services.AddScoped<PatientService>();
+builder.Services.AddScoped<DoctorService>();
+builder.Services.AddScoped<AppointmentService>();
```
Интересно, а почему же не работает ни один метод контроллера?

alxmcs commented on Hospital/.idea/.idea.Hospital/.idea/indexLayout.xml:
Похоже, что добавление строчек в .gitignore не помогло убрать файлы среды разработки из репозитория
Попробуй `.idea/*`

alxmcs commented on Hospital/Hospital.Application.Contracts/Dtos/AppointmentGetDto.cs:
<img width="1420" height="813" alt="image" src="https://github.com/user-attachments/assets/43c51812-5e66-4185-a9f9-bef23930d40c" />
Офигенно ты завел контракты и скрыл от потребителя апи доменные классы

alxmcs commented on Hospital/Hospital.Application.Contracts/Dtos/AppointmentDto.cs:
```diff
@@ -0,0 +1,39 @@
+    /// <summary>
+    /// <see cref="Doctor"/> for Appointment
+    /// </summary>
+    public required Doctor Doctor { get; set; }
```
1. Внутри dto, конечно, не запрещена агрегация. Но не доменных же сущностей.
2. Зачем тебе помещать их внутрь этого контракта целиком, если у доктора и пациента есть по своему контроллеру, откуда можно получить их данные по идентификатору? Помести их идентификаторы

alxmcs commented on Hospital/Hospital.Application.Contracts/Dtos/AppointmentGetDto.cs:
Я уже немного устал повторять, что раз ты заводишь контракты, то заведи их отдельно на get и отдельно на post/put

alxmcs commented on Hospital/Hospital.Application.Contracts/Hospital.Application.Contracts.csproj:
```diff
@@ -0,0 +1,14 @@
+        <TargetFramework>net8.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+        <GenerateDocumentationFile>true</GenerateDocumentationFile>
```
Документация контрактов все равно не подтянулась в сваггер, потому что ты не указал сборку этой библиотеки в `Hospital.Api.Program` при настройке `SwaggerGen`

alxmcs commented on Hospital/Hospital.Infrastructure.InMemory/Hospital.Infrastructure.InMemory.csproj:
```diff
@@ -0,0 +1,14 @@
+        <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <ProjectReference Include="..\Hospital.Application\Hospital.Application.csproj" />
+    </ItemGroup>
```
Зачем тебе референс к этому проекту, если `IRepository` - единственный внешний класс, который это либе нужен, и он находится в доменной библиотеке?

alxmcs commented on Hospital/Hospital.Application.Contracts/Hospital.Application.Contracts.csproj:
```diff
@@ -0,0 +1,14 @@
+
+    <ItemGroup>
+      <ProjectReference Include="..\Hospital.Domain\Hospital.Domain.csproj" />
+    </ItemGroup>
```
Стоило сразу задуматься о то, что ты что-то делаешь не так, если проект с контрактами для презентационного уровня зависим от доменной библиотеки.

alxmcs reviewed (approved):
Ну это ноль. Тут как бы и за скорость баллов прибавлять не за что - софт просто не работает

Upd: Коллегам удалось убедить меня что ноль - это слишком жестоко.

BarMan555 committed:
Fix Api controllers.

BarMan555 committed:
Update .gitignore

BarMan555 committed:
Add GetDtos and CreateUpdateDtos. Update controllers and services

BarMan555 committed:
Fix dependecies

BarMan555 committed:
Fix dtos

BarMan555 committed:
Refix services

BarMan555 committed:
Fix analytics methods
